ReactiveCocoa/Swift/Signal.swift

```diff
@@ -135,19 +146,28 @@ public final class Signal<Value, Error: ErrorProtocol> {
 	///   - observer: An observer to forward the events to.
 	///
 	/// - returns: An optional `Disposable` which can be used to disconnect the
-	///            observer. Disposing of the Disposable will have no effect on
-	///            the Signal itself.
+	///            observer.
 	@discardableResult
 	public func observe(_ observer: Observer) -> Disposable? {
 		var token: RemovalToken?
-		atomicObservers.modify { observers in
-			token = observers?.insert(observer)
+		state.modify { state in
+			if state != nil {
+				if state!.observers.isEmpty {
+					state!.retainedSignal = self
+				}
+				token = state!.observers.insert(observer)
+			}
```

Comment:
Yeah. I just put it that way for (perhaps unnecessarily) one less `nil` check and less releases & retains.